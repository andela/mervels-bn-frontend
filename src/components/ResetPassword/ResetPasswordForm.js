/* eslint-disable jsx-a11y/label-has-associated-control */
/* eslint-disable no-empty-pattern */
/* eslint-disable react/prop-types */
/* eslint-disable no-shadow */
/* eslint-disable no-unused-vars */
import React from "react";
import Input from "../shared/input";
import Button from "../shared/Button";


export const ResetFormTemplate = ({ handleChange, handleSubmit, email, submitting }) => (
  <form onSubmit={handleSubmit} name="emailForm">
    <label htmlFor="email">
      <p>
        Inorder to reset password please provide an email linked to your barefoot
        nomad account below
      </p>
       <Input
          name="email"
          inputType="email"
          placeholder="Email Address"
          onChange={handleChange}
          required
        />
    </label>
    <Button ButtonId="reset-password" submitting={submitting} buttonType="submit"  classes="btn btn-primary" text="Reset Password" />
    {/* <button className="btn btn-primary" type="submit">Reset Password</button> */}
  </form>
);

export const ResetEmailSentTemplate = ({ email }) => (
  <div>
    <p>Kindly check your email: <strong>{email}</strong> for Password reset information</p>
  </div>
);

export const ResetEmailComplete = () => (
  <div>
    <p>Password Reset Successfully.</p>
  </div>
);

export const PasswordResetFormTemplate = ({ handleChange, handleSubmit, error }) => (
  <div>
    <form onSubmit={handleSubmit} name="passwordForm">
      <h2>Password Reset Form</h2>
      <Input
        name="password"
        inputType="password"
        placeholder="Enter Password"
        onChange={handleChange}
        autoComplete="off"
        error={error.password}
        required
      />
      <Input
        name="newPassword"
        inputType="password"
        placeholder="Confirm Password"
        autoComplete="off"
        onChange={handleChange}
        required
      />
       <button className="btn btn-primary" type="submit">Submit</button>
    </form>
  </div>
);
