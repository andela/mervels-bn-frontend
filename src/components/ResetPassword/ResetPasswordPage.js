/* eslint-disable no-lone-blocks */
/* eslint-disable no-unused-expressions */
/* eslint-disable no-nested-ternary */
/* eslint-disable react-hooks/exhaustive-deps */
/* eslint-disable no-shadow */
/* eslint-disable no-debugger */
/* eslint-disable no-use-before-define */
/* eslint-disable react/prop-types */
/* eslint-disable no-unused-vars */
import React, { useState } from "react";
import { connect } from "react-redux";
import { toast } from "react-toastify";
import {
  sendResetPassword,
  resetPassword
} from "../../redux/actions/resetPassword";
import "../../styles/resetPassword.scss";
import logo from '../../logo/logo@2x.png';


import {
  ResetFormTemplate,
  ResetEmailSentTemplate,
  PasswordResetFormTemplate
} from "./ResetPasswordForm";

export function ResetPasswordPage({
  sendResetPassword,
  resetPassword,
  userId,
  userToken,
  errors,
  message,
  history
}) {
  const [email, setEmail] = useState();
  const [password, setPassword] = useState();
  const [newPassword, setNewPassword] = useState();
  const [emailSent, setEmailSent] = useState(false);

  React.useEffect(() => {
    redirect();
  }, [errors]);

  const handleChange = event => {
    switch (event.target.name) {
      case "email":
        setEmail(event.target.value);
        break;
      case "password":
        setPassword(event.target.value);
        break;
      case "newPassword":
        setNewPassword(event.target.value);
        break;
      default:
        break;
    }
  };

  const handleSubmit = event => {
    event.preventDefault();
    const formName = event.target.name;
    formName === "emailForm"
      ? sendResetPassword({ email })
      : resetPassword({ password, newPassword, userId, userToken });
  };

  function redirect() {
    if(userId === undefined){
      if (errors.message === "" && email !== undefined) {
        setEmailSent(true);
        toast.success("Reset Password Link sent");
      } else if (errors.message !== "" && email !== undefined) {
        toast.error(errors.message);
      }
    }else if(password !== undefined && errors.message !== "" ){
        toast.error(errors.message);
      }else if(password !== undefined && errors.message === "") {
        toast.success("Password Reset Successfully");
        history.push('/');
      }

  }

  return (
    <div className="login-container">
      <div className="local">
      <img src={logo} alt="logo" />
      {userId === undefined ? (
        !emailSent ? (
          <ResetFormTemplate
            handleChange={handleChange}
            handleSubmit={handleSubmit}
          />
        ) : (
          <ResetEmailSentTemplate email={email} />
        )
      ) : (
        <PasswordResetFormTemplate
          handleChange={handleChange}
          handleSubmit={handleSubmit}
        />
      )}
      </div>
    </div>
  );
}

function mapStateToProps({ resetPassword, errors }, ownProps) {
  const  userId  = (ownProps.match) ? ownProps.match.params.userId : "";
  const  userToken  = (ownProps.match) ? ownProps.match.params.userToken : "";
  return {
    userId,
    userToken,
    errors,
    message: resetPassword.message
  };
}

export default connect(
  mapStateToProps,
  { sendResetPassword, resetPassword }
)(ResetPasswordPage);

