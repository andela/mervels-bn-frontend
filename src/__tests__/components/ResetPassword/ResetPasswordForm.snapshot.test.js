/* eslint-disable no-unused-vars */
/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import renderer from "react-test-renderer";
import { shallow, mount } from 'enzyme';
import { ResetFormTemplate, ResetEmailSentTemplate, ResetEmailComplete, PasswordResetFormTemplate } from "../../../components/ResetPassword/ResetPasswordForm";


describe('Reset Password Form', () => {
    test('should test ResetForm Template', () => {
        const wrapper = shallow(
            <ResetFormTemplate
                onChange={jest.fn()}
                handleChange={jest.fn()}
                onSubmit={jest.fn()}
                required
                email="trans@email.com"
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
    test('should test Reset Email Template', () => {
        const wrapper = shallow(
            <ResetEmailSentTemplate
                email="trans@email.com"
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
    test('should test Reset Email Complete', () => {
        const wrapper = shallow(
            <ResetEmailComplete
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
    test('should test Password Reset Form Template', () => {
        const wrapper = shallow(
            <PasswordResetFormTemplate
                onChange={jest.fn()}
                onSubmit={jest.fn()}
                error={{password:undefined}}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
});