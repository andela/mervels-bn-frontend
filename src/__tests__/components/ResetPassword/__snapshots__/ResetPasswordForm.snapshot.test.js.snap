// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reset Password Form should test Password Reset Form Template 1`] = `ShallowWrapper {}`;

exports[`Reset Password Form should test Reset Email Complete 1`] = `ShallowWrapper {}`;

exports[`Reset Password Form should test Reset Email Template 1`] = `ShallowWrapper {}`;

exports[`Reset Password Form should test ResetForm Template 1`] = `ShallowWrapper {}`;
